pkg/grafana/notificationtemplate-handler.go

```diff
@@ -0,0 +1,160 @@
+package grafana
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/grafana/grafana-openapi-client-go/client/provisioning"
+	"github.com/grafana/grafana-openapi-client-go/models"
+	"github.com/grafana/grizzly/pkg/grizzly"
+)
+
+const KindAlertNotificationTemplate = "AlertNotificationTemplate"
+
+const notificationTemplatePattern = "alert-notification-templates/notificationTemplate-%s.%s"
+
+// AlertNotificationTemplateHandler is a Grizzly Handler for Grafana contactPoints
+type AlertNotificationTemplateHandler struct {
+	grizzly.BaseHandler
+}
+
+// NewAlertNotificationTemplateHandler returns a new Grizzly Handler for Grafana contactPoints
+func NewAlertNotificationTemplateHandler(provider grizzly.Provider) *AlertNotificationTemplateHandler {
+	return &AlertNotificationTemplateHandler{
+		BaseHandler: grizzly.NewBaseHandler(provider, KindAlertNotificationTemplate, false),
+	}
+}
+
+// ProxyConfigurator provides a configurator object describing how to proxy folders.
+func (h *AlertNotificationTemplateHandler) ProxyConfigurator() grizzly.ProxyConfigurator {
+	return &alertNotificationTemplateProxyConfigurator{
+		provider: h.Provider,
+	}
+}
+
+// ResourceFilePath returns the location on disk where a resource should be updated
+func (h *AlertNotificationTemplateHandler) ResourceFilePath(resource grizzly.Resource, filetype string) string {
+	filename := strings.ReplaceAll(resource.Name(), string(os.PathSeparator), "-")
+	return fmt.Sprintf(notificationTemplatePattern, filename, filetype)
+}
+
+// Prepare gets a resource ready for dispatch to the remote endpoint
+func (h *AlertNotificationTemplateHandler) Prepare(existing *grizzly.Resource, resource grizzly.Resource) *grizzly.Resource {
+	if !resource.HasSpecString("name") {
+		resource.SetSpecString("name", resource.Name())
+	}
+
+	return &resource
+}
+
+// Unprepare removes unnecessary elements from a remote resource ready for presentation/comparison
+func (h *AlertNotificationTemplateHandler) Unprepare(resource grizzly.Resource) *grizzly.Resource {
+	resource.DeleteSpecKey("version")
+	return &resource
+}
+
+func (h *AlertNotificationTemplateHandler) Validate(resource grizzly.Resource) error {
+	name, exist := resource.GetSpecString("name")
+	if resource.Name() != name && exist {
+		return fmt.Errorf("spec.name '%s' and metadata.name '%s', don't match", name, resource.Name())
```

Comment:
Would there be a benefit to creating a custom error to make handling it easier in tests + calls?